packages/apollo-federation/src/composition/utils.ts

```diff
@@ -50,13 +48,7 @@ export function mapFieldNamesToServiceName<Node extends { name: NameNode }>(
 }
 
 export function findDirectivesOnTypeOrField(
-  node: Maybe<
```

Comment:
Why did you remove the usage of `ObjectTypeDefinitionNode` and `InterfaceTypeDefinitionNode`?

Reply:
This is actually more permissive than before, as `TypeDefinitionNode` includes these and more.